### PR TITLE
add Indonesian MT STS dataset

### PR DESCRIPTION
**Pull request by henilp105**
This PR adds Indonesian dataset for STS task translated from STS English using Google Translate API.

HF link: https://huggingface.co/datasets/LazarusNLP/stsb_mt_id
Github link: https://github.com/LazarusNLP/indonesian-sentence-embeddings

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

### Adding datasets checklist

**Reason for dataset addition**: It covers Indonesian language for STS task as the `id`/`indonesian` language was not available for STS. (https://huggingface.co/datasets/LazarusNLP/stsb_mt_id) 

- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

CC: @KennethEnevoldsen @Muennighoff @LazarusNLP

### CODE REVIEW

File: docs/mmteb/points/922.jsonl [hunk 1]
```diff
@@ -0,0 +1 @@
+{"GitHub": "henilp105", "New dataset": 6}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/922.jsonl:1]: We have exceeded the deadline for dataset additions, so we have sadly stopped adding points for dataset additions (besides a few exceptions).

We would, however, welcome you to contribute to other parts of the project to obtain a co-authorship. You might get an overview as #784.

File: mteb/tasks/STS/ind/STSIDMT.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        main_score="cosine_spearman",
+        date=("2012-01-01", "2024-01-04"),  # rough estimate,
+        form=["spoken", "written"],
+        domains=["News", "Social", "Web"],
+        task_subtypes=[],
+        license="apache-2.0",
+        socioeconomic_status="mixed",
+        annotations_creators="human-annotated",
+        dialect=[],
+        text_creation="machine-translated",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/STS/ind/STSIDMT.py:35]: We do not accept machine-translated datasets unless human-verified (with few exceptions).

Comment (KennethEnevoldsen) [reply mteb/tasks/STS/ind/STSIDMT.py:35]: Note that if you e.g. want to create an Indonesian benchmark and find that a translation STS task is required we have had previous cases where we have done it, but generally we have found that machine translated datasets are not required and likely contains unknown biases.

File: docs/mmteb/points/922.jsonl (hunk 1 above)
Comment (henilp105) [reply docs/mmteb/points/922.jsonl:1]: Thanks @KennethEnevoldsen, will try contributing to other parts of the project.

File: mteb/tasks/STS/ind/STSIDMT.py (hunk 2 above)
Comment (henilp105) [reply mteb/tasks/STS/ind/STSIDMT.py:35]: Thanks @KennethEnevoldsen, No problem!

**Pull request closed without merging**